proto/redpanda/api/console/v1alpha1/security.proto

```diff
@@ -0,0 +1,236 @@
+syntax = "proto3";
+
+package redpanda.api.console.v1alpha1;
+
+import "buf/validate/validate.proto";
+
+// Role defines a role in the system.
+message Role {
+  // The name of the role.
+  string name = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+}
+
+// ListRolesRequest is the request for ListRoles.
+message ListRolesRequest {
+  // Filter options.
+  message Filter {
+    // Filter results only roles named with the prefix.
+    string name_prefix = 1 [
+      (buf.validate.field).string.max_len = 128,
+      (buf.validate.field).string.pattern = "^([a-zA-Z0-9-_]*)$"
+    ];
+
+    // Filter results to only roles with names which contain the string.
+    string name_contains = 2 [
+      (buf.validate.field).string.max_len = 128,
+      (buf.validate.field).string.pattern = "^([a-zA-Z0-9-_]*)$"
+    ];
+
+    // Return only roles assigned to this principal.
+    string principal = 3 [(buf.validate.field).string.max_len = 128];
+  }
+
+  // Optional filter.
+  optional Filter filter = 1;
+
+  // Page size.
+  int32 page_size = 2 [(buf.validate.field).int32 = {
+    gte: -1,
+    lte: 1000
+  }];
+
+  // Value of the next_page_token field returned by the previous response.
+  // If not provided, the system assumes the first page is requested.
+  string page_token = 3;
+}
+
+// ListRolesResponse is the response for ListRoles.
+message ListRolesResponse {
+  // The roles in the system.
+  repeated Role roles = 1;
+
+  // Token to retrieve the next page.
+  string next_page_token = 2;
+}
+
+// CreateRoleRequest is the request for CreateRole.
+message CreateRoleRequest {
+  // The role to create.
+  Role role = 1;
+}
+
+// CreateRoleResponse is the response for CreateRole.
+message CreateRoleResponse {
+  // The role.
+  Role role = 1;
+}
+
+// CreateRoleRequest is the request for CreateRole.
+message GetRoleRequest {
+  // The role name.
+  string role_name = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+}
+
+// GetRoleResponse is the response to GetRole.
+message GetRoleResponse {
+  // The Role.
+  Role role = 1;
+
+  // Members assigned to the role.
+  repeated RoleMembership members = 2;
+}
+
+// UpdateRoleRequest is the request for UpdateRole.
+message UpdateRoleRequest {
```

Comment:
You to update a role you provide the role_name and the role of type `role`. Type `role` however only contains a name. Is this supposed to be a role rename then only?

Reply:
Correct, currently a role only has a single property, name. So we need the existing role name, provided via `role_name` which could be used to map to a path param for HTTP API should we need to in the future. Then the content for the update is in the `Role role` field.